equipment-service/src/main/java/com/telecominfraproject/wlan/equipment/controller/EquipmentController.java

```diff
@@ -275,19 +275,37 @@ private void validateChannelNum(Equipment equipment) {
 					Integer backupChannelNum = elementRadioConfig.getBackupChannelNumber();
 					Integer manualBackupChannelNum = elementRadioConfig.getManualBackupChannelNumber();
 					
-					List<Integer> allowedChannels = elementRadioConfig.getAllowedChannelsPowerLevels().stream().map(ChannelPowerLevel::getChannelNumber).collect(Collectors.toList());
+					List<Integer> allowedChannels = elementRadioConfig.getAllowedChannelsPowerLevels().stream().
+							map(ChannelPowerLevel::getChannelNumber).collect(Collectors.toList());
 					
 					if (allowedChannels != null && !allowedChannels.isEmpty()) {
 						checkAllowedChannels(channelNum, "channelNumber", allowedChannels);
+						backupChannelNum = correctOldInvalidBackupChannel(radioType, backupChannelNum, elementRadioConfig, false);
 						checkAllowedChannels(backupChannelNum, "backupChannelNumber", allowedChannels);
+						
 						checkAllowedChannels(manualChannelNum, "manualChannelNumber", allowedChannels);
+						manualBackupChannelNum = correctOldInvalidBackupChannel(radioType, manualBackupChannelNum, elementRadioConfig, true);
 						checkAllowedChannels(manualBackupChannelNum, "manualBackupChannelNumber", allowedChannels);
 					}
 				}
 			}
 		}
 	}
 	
+	// Correct old out of range default backup channel number setting (154) on 5GU
+	private Integer correctOldInvalidBackupChannel(RadioType radioType, Integer channelNum,
+			ElementRadioConfiguration elementRadioConfig, boolean manual) {
+		if (radioType == RadioType.is5GHzU && channelNum != null && channelNum == 154) {
+			if (manual) {
+				elementRadioConfig.setManualBackupChannelNumber(157);
+			} else {
+				elementRadioConfig.setBackupChannelNumber(157);
+			}
+			return 157;
+		}
+		return channelNum;
```

Comment:
This is wrong on a number of levels.
The available channels are reported in channelPowerLevels on a per radio band. These are the channels reported as allowed by the AP, and by design this is the correct set to use.
Sort them if you are looking for the uppermost allowed channel in that band, if that is the concern, but transparently changing it in the code, to a hardcoded value that may be wrong tomorrow, is never a good idea.

Reply:
@dmitry-toptygin-connectus-ai , @mikehansen1970 , I will discard this change. We will ask users in the community to delete/re-add AP if the error happens. Thank you.